### PR TITLE
chore(lua transform): Add events to lua v2 transform.

### PR DESCRIPTION
**Pull request by Hoverbear**
Add metrics to lua v2 transform, resolving #3400 .

(Please give a bit of extra care with this review, I am not that familiar with the lua stuff.)

### CODE REVIEW

File: src/transforms/lua/v2/mod.rs [hunk 1]
```diff
@@ -329,7 +334,7 @@ impl RuntimeTransform for Lua {
 }
 
 #[cfg(test)]
-fn format_error(error: &rlua::Error) -> String {
+pub(crate) fn format_error(error: &rlua::Error) -> String {
```
Comment (lukesteensen) [inline src/transforms/lua/v2/mod.rs:337]: Why this change? Doesn't seem like we're using it anywhere new.

Comment (Hoverbear) [reply src/transforms/lua/v2/mod.rs]: I... have no idea.
